tutorials/3d/spring_arm.rst

```diff
@@ -0,0 +1,131 @@
+------------
+
+3D games will often have a third-person camera that follows and
+rotates around something such as a player character or a vehicle.
+
+In Godot, this can be done by setting a :ref:`Camera3D <class_Camera3D>` as a child of a node.
+However, if you try this without any extra steps you'll notice that the camera clips through geometry and hides the scene.
+
+This is where the :ref:`SpringArm3D <class_SpringArm3D>` node comes in.
+
+What is a spring arm?
+---------------------
+
+A spring arm has two main components that affect its behavior.
+
+The "length" of the spring arm is how far from its global position to check for collisions:
+
+.. image:: img/spring_arm_position_length.webp
+
+The "shape" of the spring arm is what it uses to check for collisions. The spring arm will "sweep" this shape from its origin out towards its length.
+
+.. image:: img/spring_arm_shape.webp
+
+The spring arm tries to keep all of its children at the end of its length. When the shape collides with something, the children are instead placed at or near that collision point:
+
+.. image:: img/spring_arm_children.webp
+
+Spring arm with a camera
+------------------------
+
+When a camera is placed as a child of a spring arm, a pyramid representing the camera will be used as the shape.
+
+This pyramid represents the **near plane** of the camera:
+
+.. image:: img/spring_arm_camera_shape.webp
+
+.. note:: If the spring arm is given a specific shape, then that shape will **always** be used.
+
+    The camera's shape is only used if the camera is a **direct child** of the spring arm.
+
+    If no shape is provided and the camera is not a direct child, the spring arm will fall back to using a ray cast which is inaccurate for camera collisions and not recommended.
+
+Every physics process frame, the spring arm will perform a motion cast to check if anything is collided with:
+
+.. image:: img/spring_arm_camera_motion_cast.webp
+
+When the shape hits something, the camera will be placed at or near the collision point:
+
+.. image:: img/spring_arm_camera_collision.webp
+
+Setting up the spring arm and camera
+------------------------------------
+
+Let's add a spring arm camera setup to the platformer demo.
+
+.. note:: You can download the Platformer 3D demo on `GitHub <https://github.com/godotengine/godot-demo-projects/tree/master/3d/platformer>`_ or using the `Asset Library <https://godotengine.org/asset-library/asset/2748>`_.
+
+In general, for a third-person camera setup, you will have three nodes as children of the node that you're following:
+
+- `Node3D` (the "pivot point" for the camera)
+
+    - `SpringArm3D`
+
+        - `Camera3D`
+
+Open the ``player/player.tscn`` scene. Set these up as children of our player and give them unique names so we can find them in our script. **Make sure to delete the existing camera node!**
+
+.. image:: img/spring_arm_editor_setup.webp
+
+Let's move the pivot point up by ``3`` on the Y-axis so that it's not on the ground:
```

Comment:
```suggestion
Let's move the pivot point up by ``2`` on the Y-axis so that it's not on the ground:
```
(And change the screenshot if you can)

Nitpick, but a value of `3` puts the camera pivot solidly above the head of the player:

![Godot_v4 3-stable_win64_fuIwdhkgeo](https://github.com/user-attachments/assets/9458eddb-f170-4277-9b6d-19694f6811ff)

If you stand close to the wall and try to clip the camera into the wall, instead you don't see the player at all:

https://github.com/user-attachments/assets/cb498802-8fb9-49e0-bfc2-77c8256e7b2f

Reply:
**y = 2**
![Godot_v4 3-stable_win64_RDCBcYZr27](https://github.com/user-attachments/assets/31eadeaa-5a33-4729-95f7-ebc84f1e7d3b)

**y = 3**
![Godot_v4 3-stable_win64_Ap5XPJDF0A](https://github.com/user-attachments/assets/0c4276fe-fde9-4059-a7ad-6e371e9a7161)

Maybe there's some other difference in setup. I imported the demo into 4.3, then made the changes as suggested by your tutorial. The CameraPivot is a direct child of the Player node, and it has a position of either `(0,2,0) or `(0,3,0)`.

This isn't a huge deal either way. I just noticed because I tried to recreate the video at the end of the tutorial, and the pivot was too far off the ground to actually see the player as the camera rotated. I could still see how the spring arm generally works.

Reply:
Okay, you're totally right! I had some other changes that were affecting it

Reply:
Okay, updated it to be `2` instead of `3` 👍